stdlib/array.gr

```diff
@@ -1015,6 +1015,53 @@ provide let rotate = (n, arr) => {
   }
 }
 
+let rec _chunk = (chunkSize, allowPartial, arr, arrLen, chunkStartIndex) => {
+  if (arrLen == 0 || arrLen < chunkSize && !allowPartial) {
+    []
+  } else if (arrLen < chunkSize) {
+    [slice(chunkStartIndex, arr)]
+  } else {
+    // create the first chunk of the given array
+    let firstChunk = slice(
+      chunkStartIndex,
+      end=chunkStartIndex + chunkSize,
+      arr
+    )
+    let restChunks = _chunk(
+      chunkSize,
+      allowPartial,
+      arr,
+      arrLen - chunkSize,
+      chunkStartIndex + chunkSize
+    )
+    [firstChunk, ...restChunks]
+  }
+}
+
+/**
+ * Splits the given array into chunks of the provided size.
+ *
+ * @param chunkSize: The size of each chunk
+ * @param arr: The array to chunk
+ * @returns An array of chunks
+ *
+ * @throws InvalidArgument(String): When `chunkSize` is not an integer
+ * @throws InvalidArgument(String): When `chunkSize` is less than zero
```

Comment:
```suggestion
 * @throws InvalidArgument(String): When `chunkSize` is less than one
```
I'm not sure about this. You check explicitly for `0` but you then use slice, which accepts negative indexing. Do you need different checks?